### PR TITLE
Fix jar name & bump to v0.31.0-alpha3

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Detailed description**:
- Fix Rosetta validation failing on tags
- Fix release jar not prefixed with `v`
- Bump versions to v0.31.0-alpha3

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Review by Nana-EC — APPROVED** [review]
LGTM

**Review by ijungmann — DISMISSED** [review]

**Review by xin-hedera — APPROVED** [review]

File: .github/workflows/rosetta-api.yml [hunk 1]
```diff
@@ -53,7 +53,7 @@ jobs:
       - uses: actions/checkout@v2
 
       - name: Build Mirror Node All-in-One
-        run: docker build --tag="${MODULE}" --build-arg GIT_BRANCH=${{ github.head_ref }} build/
```
Comment (xin-hedera) [inline .github/workflows/rosetta-api.yml:56]: still failing

**Review by xin-hedera — CHANGES_REQUESTED** [review]

File: .github/workflows/rosetta-api.yml (hunk 1 above)
Comment (xin-hedera) [reply .github/workflows/rosetta-api.yml:56]: `github.ref` should work

Comment (steven-sheehy) [reply .github/workflows/rosetta-api.yml:56]: nope

Comment (xin-hedera) [reply .github/workflows/rosetta-api.yml:56]: the `sha` is sth not even in the git repo... wth :(

Comment (xin-hedera) [reply .github/workflows/rosetta-api.yml:56]: FYI: https://github.com/actions/checkout/issues/27

Comment (steven-sheehy) [reply .github/workflows/rosetta-api.yml:56]: I think this solution works well enough. Bit verbose but should suffice.

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**